src/services/reportService.ts

```diff
@@ -0,0 +1,27 @@
+import reportRepository from '../repositories/reportRepository';
+
+const selectProductsFromService = async ({
+  limit = '100',
+  orderSort = 'DESC',
+  orderField = 'sale_competitors_month',
+  category = '',
+  period = ''
+}) => {
+  const whereQuery = reportRepository.whereConstructor({
+    category: category,
+    period: period
+  });
+
+  const orderQuery = reportRepository.orderConstructor({
+    orderField: orderField,
+    orderSort: orderSort
+  });
+  const limitAsNumber = typeof limit === 'string' ? parseInt(limit) : 100;
```

Comment:
hmmmm ... 🤨

Vamos refletir um pouco aqui ...
1 - Porque limit está sendo iniciado com uma string?  `'100'`
2 - Percebe que nessa verificação o limitAsNumber acaba ignorando valores inteiros, porque se for uma string transforma para um numero, mas se não for string coloca 100 como valor padrão?
3 - Parece que a ideia é colocar 100 como valor padrão ... como já está sendo iniciado com esse valor, será que ele precisa ser o `else` no ternário?

Reply:
Resolvido também :) muito obrigado